src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User must not be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("A user with this login is already registered");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("You are not have 18 years old");
```

Comment:
The error message for the age validation is not grammatically correct and could be more informative. Consider rephrasing it to something like 'User must be at least 18 years old.' Also, split the age validation into separate `if` statements to handle `null` and the age check individually.